californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -219,8 +219,10 @@ public void receiveRequest(final Exchange exchange, final Request request) {
 					if (request.isConfirmable()) {
 						// resend ACK,
 						// comply to RFC 7252, 4.2, cross-layer behavior
-						EmptyMessage ack = EmptyMessage.newACK(request);
-						sendEmptyMessage(exchange, ack);
+						if (request.acknowledge()) {
+							EmptyMessage ack = EmptyMessage.newACK(request);
+							sendEmptyMessage(exchange, ack);
+						}
```

Comment:
reading this, it seems to me that we always send an empty ACK even if we replied with a piggyback, but I suppose I miss read the code ?

Reply:
Piggy-backed response have type ACK, and will bypass this.
If the response was NON, usually a ACK was sent, so this just retransmit the ACK as well.

Reply:
I get it.